src/rigid_pendulum_poc/generalized_alpha_time_integrator.cpp

```diff
@@ -106,111 +109,125 @@ std::tuple<State, HostView1D> GeneralizedAlphaTimeIntegrator::AlphaStep(
             "Iteration number: " + std::to_string(time_stepper_.GetNumberOfIterations() + 1) + "\n"
         );
 
-        auto residuals = ComputeResiduals(gen_coords, vector);
-        // TODO: Provide actual force increments
-        auto increments = residuals;
+        auto gen_coords_next = ComputeUpdatedGeneralizedCoordinates(gen_coords, x);
 
-        if (this->CheckConvergence(residuals, increments)) {
+        // Compute the residuals and check for convergence
+        auto residuals = ComputeResiduals(acceleration, mass_matrix, gen_forces);
+        if (this->CheckConvergence(residuals)) {
             this->is_converged_ = true;
             break;
         }
 
+        // Compute the iteration matrix and solve the linear system to get the increments
         auto iteration_matrix = ComputeIterationMatrix(gen_coords, matrix);
-        auto gen_coords_delta = residuals;
-        auto constraints_delta = constraints;
-        solve_linear_system(iteration_matrix, gen_coords_delta);
-        solve_linear_system(iteration_matrix, constraints_delta);
+        auto delta_x = residuals;
+        auto delta_constraints = constraints;
 
+        solve_linear_system(iteration_matrix, delta_x);
```

Comment:
This must be a single-system solve -- the iteration matrix is not block diagonal.  See Eq. (34) in Bruls et al 2012